### PR TITLE
[CODEC-295] - Minor improvements

### PR DESCRIPTION
**Pull request by arturobernalg**
https://issues.apache.org/jira/browse/CODEC-295

Minor improvement 

- extra computations by getting the string size only once.
- remove unused imports
- add final String
- add final method and var
- FIx javadoc

### CODE REVIEW

**Comment by arturobernalg** [comment]
HI @garydgregory  @sebbASF  any idea of why the Java CI  it's failing?

**Comment by garydgregory** [comment]
@arturobernalg 
Please rebase on master.

File: src/main/java/org/apache/commons/codec/binary/Hex.java [hunk 1]
```diff
@@ -168,7 +168,7 @@ public static int decodeHex(final char[] data, final byte[] out, final int outOf
     protected static char[] encodeHex(final byte[] data, final char[] toDigits) {
         final int l = data.length;
         final char[] out = new char[l << 1];
-        encodeHex(data, 0, data.length, toDigits, out, 0);
+        encodeHex(data, 0, l, toDigits, out, 0);
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/codec/binary/Hex.java:171]: No mystery names please, especially "l" which can look just like a "1" depending on your font!

File: src/main/java/org/apache/commons/codec/cli/Digest.java [hunk 2]
```diff
@@ -61,16 +61,17 @@ private Digest(final String[] args) {
...
                     String.format("Usage: java %s [algorithm] [FILE|DIRECTORY|string] ...", Digest.class.getName()));
         }
         this.args = args;
         algorithm = args[0];
-        if (args.length <= 1) {
+        if (argsLength <= 1) {
             inputs = null;
         } else {
-            inputs = new String[args.length -1];
+            inputs = new String[argsLength -1];
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/codec/cli/Digest.java:74]: Formatting.

File: src/main/java/org/apache/commons/codec/language/bm/Rule.java [hunk 3]
```diff
@@ -488,11 +496,12 @@ private static RPattern pattern(final String regex) {
         final boolean endsWith = regex.endsWith("$");
         final String content = regex.substring(startsWith ? 1 : 0, endsWith ? regex.length() - 1 : regex.length());
         final boolean boxes = content.contains("[");
+        final int contentLength = content.length();
 
         if (!boxes) {
             if (startsWith && endsWith) {
                 // exact match
-                if (content.length() == 0) {
+                if (contentLength == 0) {
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/codec/language/bm/Rule.java:504]: Just use `String#isEmpty()` for 0-length checks on Strings.

File: src/main/java/org/apache/commons/codec/language/bm/Rule.java [hunk 4]
```diff
@@ -507,7 +516,7 @@ public boolean isMatch(final CharSequence input) {
                         return input.equals(content);
                     }
                 };
-            } else if ((startsWith || endsWith) && content.length() == 0) {
+            } else if ((startsWith || endsWith) && contentLength == 0) {
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/codec/language/bm/Rule.java:519]: See above.

File: src/test/java/org/apache/commons/codec/cli/DigestTest.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+ * @since 1.17
+ */
+public class DigestTest{
+
+    /**
+     * Tests if empty arguments are handled correctly.
+     *
+     * @throws IllegalArgumentException for some failure scenarios.
+     */
+    @Test(expected = IllegalArgumentException.class)
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/codec/cli/DigestTest.java:37]: Do we need a local var?

**Review by garydgregory — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/commons/codec/binary/Hex.java (hunk 1 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/codec/binary/Hex.java]: agree. fixit

File: src/main/java/org/apache/commons/codec/cli/Digest.java (hunk 2 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/codec/cli/Digest.java]: agree. fixit

File: src/main/java/org/apache/commons/codec/language/bm/Rule.java (hunk 3 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/codec/language/bm/Rule.java]: agree. fixit

File: src/test/java/org/apache/commons/codec/cli/DigestTest.java (hunk 5 above)
Comment (arturobernalg) [reply src/test/java/org/apache/commons/codec/cli/DigestTest.java:37]: agree. fixit

File: src/test/java/org/apache/commons/codec/cli/DigestTest.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+import java.io.IOException;
+
+
+/**
+ * Tests {@link Digest}.
+ *
+ * @since 1.17
+ */
+public class DigestTest{
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/codec/cli/DigestTest.java:30]: Fix formatting.

Comment (arturobernalg) [reply src/test/java/org/apache/commons/codec/cli/DigestTest.java]: Fix it

**Comment by garydgregory** [comment]
@arturobernalg 
IMO, we do not need Jira tickets for these kinds of changes.

**Pull request merged**